app/controllers/portal/client_logins_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Portal
+  class ClientLoginsController < ApplicationController
+    before_action :skip_login_if_authenticated
```

Comment:
Non-blocker: I suggest renaming this to `redirect_to_portal_if_client_authenticated` or something. I found it a little confusing, but anyway I get it now.